.github/workflows/simulation-1.2.yml

```
@@ -76,15 +76,26 @@ jobs:
         ./script/test prepare_coredump_upload
         ./script/test unit
         ./script/test cert_suite tests/scripts/thread-cert/v1_2_*
-    - name: Copy Shared Libraries
+    - name: Check Crash
       if: ${{ failure() }}
       run: |
-          ./script/test copy_so_lib
+          CRASHED=$(./script/test check_crash | tail -1)
+          [[ $CRASHED -eq "1" ]] && echo "Crashed!" || echo "Not crashed."
+          echo "CRASHED=$CRASHED" >> $GITHUB_ENV
     - uses: actions/upload-artifact@v2
       if: ${{ failure() }}
       with:
         name: thread-1-2-${{ matrix.compiler.c }}-${{ matrix.arch }}-pcaps
         path: "*.pcap"
+    - uses: actions/upload-artifact@v2
+      if: ${{ failure() && env.CRASHED == '1' }}
+      with:
+        name: core-packet-verification-low-power
+        path: |
+          ./build/openthread-simulation-1.2/examples/apps/cli/ot-cli-*
+          ./build/openthread-simulation-1.1/examples/apps/cli/ot-cli-*
+          ./ot-core-dump/*
+          ./so-lib/*
```

```suggestion
    - uses: actions/upload-artifact@v2
      if: ${{ failure() }}
      with:
        name: core-packet-verification-low-power
        path: |
          ./ot-core-dump/*
```
Can we copy everything we need to `ot-core-dump/` (including ot-cli-* executables and so-lib/*) when core dumped and clear ot-core-dump otherwise? So that we automatically uploads nothing if there was no core dump.

---

Put everything into one dir SGTM. While I still prefer to use the env variable. In this way, the upload action won't be started at all (Otherwise, we can still see the action processed, and it fails to find anything in the dir). I think that would be a stronger signal telling us that there's no crash. Thoughts?

---

I was hoping that this way we can make the github actions jobs simpler --- we only need to run `./script/test check_crash` and upload one artifact directory (i.e. ./ot-core-dump/) after that.
Thoughts?

---

I mean I agree to put everything into one directory. And then in `script/test` we can use `file` command to scan the core dump files and collect what binaries crashed, putting them into the directory. In this way we can make these jobs in simplest way.

My point is just that we can skip the unnecessary upload action (which is apparently faster than running it though not uploading anything) by:
```
- if: ${{ failure() }}
+ if: ${{ failure() && env.CRASHED == '1'}}
```

So the line numbers don't change. Thoughts?

---

This approach SGTM.